sqlmesh/utils/date.py

```diff
@@ -273,3 +274,15 @@ def time_like_to_str(time_like: TimeLike) -> str:
     if is_date(time_like):
         return to_ds(time_like)
     return to_datetime(time_like).isoformat()
+
+
+def to_end_date(
+    end_and_units: t.Union[t.Tuple[int, IntervalUnit], t.List[t.Tuple[int, IntervalUnit]]]
+) -> TimeLike:
+    from sqlmesh.core.model.meta import IntervalUnit
+
+    end_and_units = [end_and_units] if isinstance(end_and_units, tuple) else end_and_units
+    end, unit = max(end_and_units)
+    if unit == IntervalUnit.DAY:
```

Comment:
CC: @crericha I assume we'd need the same thing for `MONTH`, `WEEK` and `YEAR` units.